### PR TITLE
[Core] Distance processes 2D compatible

### PR DESCRIPTION
**Pull request by rubenzorrilla**
This PR finally makes the distance computation processes 2D compatible. What I've done is to consider the 2D as a particular case of the 3D one. This implies that even though we are in a 2D space, we still use the binary octree for the distance computation. I've added several tests (multiple simple cpp tests and a Python aerofoil one) so you can check how is the usage. Hopefully, the previous behaviour shouldn't be affected, but please report any strange behaviour.

Btw, I took the chance to do a major clean up and to add documentation to both continuous and discontinuous distance processes.

### CODE REVIEW

File: kratos/processes/calculate_distance_to_skin_process.cpp [hunk 1]
```diff
@@ -7,215 +7,228 @@
...
 	}
 
-	void CalculateDistanceToSkinProcess::CalculateDistances(std::vector<PointerVector<GeometricalObject>>& rIntersectedObjects)
+	template<std::size_t TDim>
+	void CalculateDistanceToSkinProcess<TDim>::CalculateDistances(
+		std::vector<PointerVector<GeometricalObject>>& rIntersectedObjects)
 	{
-		this->CalculateElementalDistances(rIntersectedObjects);
+		// Compute the discontinuous (elemental) distance field
+		const bool use_base_elemental_distance = false;
```
Comment (rubenzorrilla) [inline kratos/processes/calculate_distance_to_skin_process.cpp:63]: @pooyan-dadvand I though that it is a good idea to let the use of both the base and the simple (implemented in this class) elemental distance computation. The use of the naive one (without plane optimization) is hardcoded. This is the one that we've used historically and it yields smoother continuous distance fields according to the preliminary tests I've done. Tell me if you have any inconvenience on that.

**Comment by philbucher** [comment]
@adityaghantasala @rishithellathmeethal this might improve the hole-cutting

**Comment by adityaghantasala** [comment]
@philbucher  I will have to check. I remember its not about distance calculation itself but more about how the rays are shot for finding inside and outside.

**Comment by roigcarlo** [comment]
Also ping to @playpaolo. He is very interested and may want to test

**Comment by playpaolo** [comment]
great @roigcarlo ! Will test it

**Review by roigcarlo — DISMISSED** [review]
:+1:

**Comment by roigcarlo** [comment]
But there are some conflicts and you need to take a of a couple of warnings

**Comment by rubenzorrilla** [comment]
@roigcarlo the conflicts arose because of the Model... Hopefully they will be fixed with the last changes.

File: kratos/processes/calculate_distance_to_skin_process.cpp (hunk 1 above)
Comment (pooyan-dadvand) [reply kratos/processes/calculate_distance_to_skin_process.cpp:63]: I see, agreed 👍

File: kratos/processes/calculate_distance_to_skin_process.cpp [hunk 2]
```diff
@@ -7,215 +7,228 @@
...
-        intersections_container_type intersections;
-
-        const int dimension = 3;
         const double epsilon = 1e-12;
+		array_1d<double,TDim> distances;
+        intersections_container_type intersections;
 
-        double distances[3] = {1.0, 1.0, 1.0};
+		// Loop the x,y and z (3D) ray directions
+        for (int i_direction = 0; i_direction < TDim; i_direction++){
```
Comment (rubenzorrilla) [inline kratos/processes/calculate_distance_to_skin_process.cpp]:
```suggestion
        for (unsigned int i_direction = 0; i_direction < TDim; i_direction++){
```

**Review by roigcarlo — APPROVED** [review]
:+1:

**Review by roigcarlo — APPROVED** [review]
:+1:

**Review by roigcarlo — APPROVED** [review]

**Review by roigcarlo — APPROVED** [review]

**Comment by roigcarlo** [comment]
mm seems that I over-approved this.

**Pull request merged**